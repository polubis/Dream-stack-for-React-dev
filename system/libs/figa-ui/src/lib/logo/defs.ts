import type { ReactNode } from 'react';

interface LogoProps {
  className?: string;
  graphic?: ReactNode;
  parts?: [string, string, string];
}

export type { LogoProps };
