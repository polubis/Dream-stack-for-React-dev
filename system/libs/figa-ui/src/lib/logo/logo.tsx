import { LogoGraphic } from '../logo-graphic';
import type { LogoProps } from './defs';

import c from 'classnames';

const Logo = ({
  className,
  graphic = <LogoGraphic size={52} />,
  parts = ['Green', 'On', 'Software'],
}: LogoProps) => {
  return (
    <div className={c('logo', className)}>
      {graphic}
      <div className="logo-text">
        <div className="logo-text-top">
          <span>{parts[0]}</span>
          <span>{parts[1]}</span>
        </div>
        <span className="logo-text-bottom">{parts[2]}</span>
      </div>
    </div>
  );
};

export { Logo };
