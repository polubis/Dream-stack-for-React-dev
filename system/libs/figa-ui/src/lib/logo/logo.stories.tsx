import type { Story, Meta } from '@storybook/react';

import { Logo } from './logo';

export default {
  component: Logo,
  title: 'Logo',
} as Meta;

const Template: Story = () => {
  return (
    <div style={{ display: 'flex', flexFlow: 'wrap', gap: '32px' }}>
      <Logo />
      <Logo parts={['First', 'Second', 'Last']} />
    </div>
  );
};

export const Default = Template.bind({});
Default.args = {};
