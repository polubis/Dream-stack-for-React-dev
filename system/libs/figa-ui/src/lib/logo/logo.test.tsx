import { render, screen } from '@testing-library/react';

import { Logo } from './logo';

describe('Logo can be used when', () => {
  it('[FRAGILE] renders with default setup', () => {
    const { asFragment } = render(<Logo />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] has the corresponding classes generated', () => {
    const { container } = render(<Logo className="my-class" />);

    const logo = container.querySelector('.logo');

    expect(logo?.className).toBe('logo my-class');
  });

  it('renders company name by default', () => {
    render(<Logo className="my-class" />);

    screen.getByText(/Green/);
    screen.getByText(/On/);
    screen.getByText(/Software/);
  });

  it('allows to pass custom text', () => {
    render(<Logo className="my-class" parts={['My', 'Custom', 'App']} />);

    screen.getByText(/My/);
    screen.getByText(/Custom/);
    screen.getByText(/App/);
  });

  it('allows to pass custom graphic node', () => {
    render(<Logo graphic={<div>My graphic</div>} />);

    screen.getByText(/My graphic/);
  });
});
