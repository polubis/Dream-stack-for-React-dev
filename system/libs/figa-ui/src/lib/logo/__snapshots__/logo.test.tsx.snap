// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo can be used when [FRAGILE] renders with default setup 1`] = `
<DocumentFragment>
  <div
    class="logo"
  >
    <svg
      fill="none"
      height="52"
      viewBox="0 0 52 52"
      width="52"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M39.3784 12.1375C44.1761 16.3698 50.2363 19.057 50.9307 22.819C51.6252 26.5811 46.9537 31.418 43.3554 35.919C39.7572 40.3528 37.2321 44.3836 33.255 47.2723C29.3411 50.0938 23.9122 51.7061 18.6095 50.6984C13.3067 49.6236 8.06716 45.9959 4.7214 40.7559C1.43877 35.4487 0.0499603 28.5964 1.69128 22.9534C3.33259 17.3103 8.00403 12.8765 12.7386 8.57704C17.5363 4.27756 22.5234 0.246812 26.7529 1.12014C30.9825 1.99347 34.5176 7.83806 39.3784 12.1375Z"
        stroke="url(#paint0_linear_102_163)"
        stroke-width="2"
      />
      <path
        d="M36.0338 15.6032C39.6321 18.7774 44.1773 20.7927 44.6981 23.6143C45.2189 26.4358 41.7153 30.0635 39.0166 33.4392C36.3179 36.7646 34.4241 39.7877 31.4413 41.9542C28.5058 44.0704 24.4341 45.2796 20.4571 44.5238C16.4801 43.7177 12.5504 40.9969 10.041 37.0669C7.57907 33.0865 6.53747 27.9473 7.76846 23.715C8.99944 19.4828 12.503 16.1574 16.0539 12.9328C19.6522 9.70817 23.3925 6.68511 26.5647 7.34011C29.7368 7.9951 32.3882 12.3785 36.0338 15.6032Z"
        stroke="url(#paint1_linear_102_163)"
        stroke-width="2"
      />
      <path
        d="M32.6892 19.0688C35.0881 21.1849 38.1182 22.5285 38.4654 24.4095C38.8126 26.2905 36.4769 28.709 34.6777 30.9595C32.8786 33.1764 31.616 35.1918 29.6275 36.6361C27.6706 38.0469 24.9561 38.8531 22.3047 38.3492C19.6534 37.8118 17.0336 35.9979 15.3607 33.3779C13.7194 30.7244 13.025 27.2982 13.8456 24.4767C14.6663 21.6552 17.002 19.4383 19.3693 17.2885C21.7681 15.1388 24.2617 13.1234 26.3765 13.5601C28.4912 13.9967 30.2588 16.919 32.6892 19.0688Z"
        stroke="url(#paint2_linear_102_163)"
        stroke-width="2"
      />
      <defs>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear_102_163"
          x1="1"
          x2="51"
          y1="51"
          y2="1"
        >
          <stop
            class="logo-bg-first-from"
          />
          <stop
            class="logo-bg-first-to"
            offset="1"
          />
        </lineargradient>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear_102_163"
          x1="7.25"
          x2="44.75"
          y1="44.75"
          y2="7.25"
        >
          <stop
            class="logo-bg-second-from"
          />
          <stop
            class="logo-bg-second-to"
            offset="1"
          />
        </lineargradient>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint2_linear_102_163"
          x1="13.5"
          x2="38.5"
          y1="38.5"
          y2="13.5"
        >
          <stop
            class="logo-bg-last-from"
          />
          <stop
            class="logo-bg-last-to"
            offset="1"
          />
        </lineargradient>
      </defs>
    </svg>
    <div
      class="logo-text"
    >
      <div
        class="logo-text-top"
      >
        <span>
          Green
        </span>
        <span>
          On
        </span>
      </div>
      <span
        class="logo-text-bottom"
      >
        Software
      </span>
    </div>
  </div>
</DocumentFragment>
`;
